exporter/datadogexporter/factory.go

```diff
@@ -264,3 +274,51 @@ func (f *factory) createTracesExporter(
 		exporterhelper.WithShutdown(stop),
 	)
 }
+
+// createLogsExporter creates a trace exporter based on this config.
+func (f *factory) createLogsExporter(
+	ctx context.Context,
+	set component.ExporterCreateSettings,
+	c config.Exporter,
+) (component.LogsExporter, error) {
+	cfg := checkAndCastConfig(c)
+
+	var pusher consumer.ConsumeLogsFunc
+	hostProvider, err := f.SourceProvider(set.TelemetrySettings, cfg.Hostname)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build hostname provider: %w", err)
+	}
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Seems like this is repeated in every exporter with the same parameters. Perhaps it should run only once and be a field on the `*factory` as `hostProvider`

Reply:
Agree , we should refactor . We can do this in a follow PR